src/shop.c

```diff
@@ -0,0 +1,1368 @@
+static void BuyMenuAddScrollIndicatorArrows(void);
+static void BuyQuantityAddScrollIndicatorArrows(void);
+static void BuyMenuRemoveScrollIndicatorArrows(void);
+static void sub_809B764(void);
+static void BuyMenuDrawMapBg(void);
+static void BuyMenuDrawMapMetatile(s16 x, s16 y, const u16 *src, u8 metatileLayerType);
+static void BuyMenuDrawMapMetatileLayer(u16 *dest, s16 offset1, s16 offset2, const u16 *src);
+static void BuyMenuCollectEventObjectData(void);
+static void BuyMenuDrawEventObjects(void);
+static void BuyMenuCopyTilemapData(void);
+static void BuyMenuPrintItemQuantityAndPrice(u8 taskId);
+static void Task_BuyMenu(u8 taskId);
+static void Task_BuyHowManyDialogueInit(u8 taskId);
+static void Task_BuyHowManyDialogueHandleInput(u8 taskId);
+static void CreateBuyMenuConfirmPurchaseWindow(u8 taskId);
+static void BuyMenuTryMakePurchase(u8 taskId);
+static void BuyMenuSubtractMoney(u8 taskId);
+static void Task_ReturnToItemListAfterItemPurchase(u8 taskId);
+static void BuyMenuReturnToItemList(u8 taskId);
+static void ExitBuyMenu(u8 taskId);
+static void Task_ExitBuyMenu(u8 taskId);
+static void nullsub_52(u8 taskId);
+static void nullsub_53(void);
+static void RecordQuestLogItemPurchase(void);
+
+extern u8 MapGridGetMetatileLayerTypeAt(s16 x, s16 y);
+extern u16 BagGetQuantityByItemId(u16 item);
+
+static const struct MenuAction sShopMenuActions_BuySellQuit[] =
+{
+    {gText_ShopBuy, {.void_u8 = Task_HandleShopMenuBuy}},
+    {gText_ShopSell, {.void_u8 = Task_HandleShopMenuSell}},
+    {gText_ShopQuit, {.void_u8 = Task_HandleShopMenuQuit}}
+};
+
+static const struct YesNoFuncTable sShopMenuActions_BuyQuit[] =
+{
+    BuyMenuTryMakePurchase,
+    BuyMenuReturnToItemList
+};
+
+static const struct WindowTemplate sShopMenuWindowTemplates[] =
+{
+    {
+        .bg = 0,
+        .tilemapLeft = 2,
+        .tilemapTop = 1,
+        .width = 12,
+        .height = 6,
+        .paletteNum = 15,
+        .baseBlock = 8
+    }
+};    
+
+static const struct BgTemplate sShopBuyMenuBgTemplates[4] =
+{
+    {
+        .bg = 0,
+        .charBaseIndex = 2,
+        .mapBaseIndex = 31,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 0,
+        .baseTile = 0
+    },
+    {
+        .bg = 1,
+        .charBaseIndex = 0,
+        .mapBaseIndex = 30,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 1,
+        .baseTile = 0
+    },
+    {
+        .bg = 2,
+        .charBaseIndex = 0,
+        .mapBaseIndex = 29,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 2,
+        .baseTile = 0
+    },
+    {
+        .bg = 3,
+        .charBaseIndex = 0,
+        .mapBaseIndex = 28,
+        .screenSize = 0,
+        .paletteMode = 0,
+        .priority = 3,
+        .baseTile = 0
+    }
+};
+
+// Functions
+static u8 CreateShopMenu(u8 a0)
+{
+    gShopData.martType = GetMartTypeFromItemList(a0) & 0xF;
+    gShopData.selectedRow = 0;
+    if (!(ContextNpcGetTextColor()))
```

Comment:
if the return value is not bool(8|16|32), write the expression as `ContextNpcGetTextColor() == 0`